python/TestHarness/schedulers/QueueManager.py

```diff
@@ -25,565 +36,322 @@ def validParams():
-        if tester.specs.isValid('gold_dir'):
-            copy_files.update([tester.specs['gold_dir']])
+        # Build [<args>, '--spec-file' ,/path/to/tests', '-o', '/path/to']
+        args.extend(['--spec-file', os.path.join(job.getTestDir(), self.options.input_file_name),
+                     '-o', job.getTestDir()])
 
-        # convert the copy and nocopy sets to flat lists
-        template['copy_files'] = list(copy_files)
-        template['no_copy'] = list(no_copy_files)
+        # Build [<command>, <args>]
+        command.extend(args)
 
-        # The location of the resulting execution script
-        template['queue_script'] = template['job_name'] + '.sh'
+        # Add the second results file we should remove when performing cleanup operations
+        self.addDirtyFiles(job, [os.path.join(job.getTestDir(), self.__job_storage_file)])
 
-        # Call derived augmentQueueParams to fill in specifics
-        template = self.augmentQueueParams(job, template)
+        return command
 
-        return template
-
-    def reserveSlots(self, job):
+    def _isProcessReady(self, job_data):
         """
-        Inherited method which controls when jobs are allowed to execute,
-        depending on available resources.
-
-        QueueManager only executes third party queueing commands. So
-        modify every job to only require 1 process.
+        Return bool on `run_tests --spec_file` submission results being available. Due to the
+        way the TestHarness writes to this results file (when the TestHarness exits), this file
+        when available, means every test contained therein is finished in some form or another.
         """
-        job.setSlots(1)
-        return Scheduler.reserveSlots(self, job)
-
-    def testOutput(self, job):
+        # Immediate reason to return false (no session file)
+        if not job_data.json_data:
+            return False
+
+        is_ready = True
+
+        # job(s) were launched but results file does not yet exist (QUEUED)
+        if (job_data.json_data.get(job_data.job_dir, {})
+            and not os.path.exists(os.path.join(job_data.job_dir, self.__job_storage_file))):
+            for job in job_data.jobs.getJobs():
+                tester = job.getTester()
+                if tester.isNoStatus():
+                    tester.setStatus(tester.no_status, 'QUEUED')
+                job.setStatus(job.finished)
+            is_ready = False
+
+        # job(s) were specifically silently skipped during initial launch
+        if not job_data.json_data.get(job_data.job_dir, {}):
+            self._finishAllJobs(job_data)
+            is_ready = False
+
+        # results file exists, but job was launched using a different scheduler
+        if (job_data.json_data.get(job_data.job_dir, {})
+            and not job_data.json_data[job_data.job_dir].get(job_data.plugin, {})):
+            self._finishAllJobs(job_data)
+            is_ready = False
+
+        return is_ready
```

Comment:
refactor this method a bit more. if elif elif, or just return False within each to same cpu cycles.